### ISSUE TITLE
Possible to compute average ATT(g,t) for custom list of pairs g-t pairs?

### ISSUE THREAD

**Issue by jtorcasso** [bcallaway11/did#151]
Sometimes I'd like to break out impacts by collections of groups (multiple, but not all, values of g) for a fixed range of event time. Is it possible to do this in the `did` package? For instance, suppose I wanted the average ATT(g,t) for g,t in  [(g1,t1),(g1,t2),(g2,t2),(g2,t3)].

**Comment by bcallaway11** [comment]
We can do this directly for some cases (e.g., you can check the `balance_e` argument of `aggte`).  If you want to include fully custom groups, that might take some more work but could be possible, I think.

Brant

**Comment by jtorcasso** [comment]
I see. That makes a lot of sense. May be useful to add an example like this in the docs somewhere. Let me walk through the logic. So suppose I have monthly data from 2019-01 through 2022-12 with staggered treatment taking place over the last three years (2020, 2021 and 2022). I want to get the average effect for months 7 through 12 in event time for only those units who got treated in 2021. Suppose `MP` is the output from `att_gt` where I run it on a dataset that includes only never treated and those units with a launch year >= 2021. I think I would then specify

```
aggte(MP, balance_e=13, min_e=7, max_e=12)
```
The sample restriction removes launches earlier than 2021. The `balance_e=13` argument ensures I have at least 13 months of post treatment data, which excludes the units treated in 2022 (which would have, at most, 12 months of post treatment data).

**Comment by jtorcasso** [comment]
Actually. This doesn't work. If I have to set `balance_e=9` and I still want the average from 7 through 12, aggte will cap the event time to 9, not 12. So something is restricting `max_e <=balance_e`.

**Comment by bcallaway11** [comment]
Yes, what you are saying here makes sense to me.  The intended use of `balance_e` is to have the compositions of groups be the same across different event times.  This is why it truncates the event times to be those $<=$ whatever the value of `balance_e` is as, otherwise, the composition will change.

Does this line: `aggte(MP, type="dynamic", balance_e=13, min_e=7, max_e=12)` work as expected?

I think the second case you are talking about will probably require some kind of custom wrapper on top of the results from `att_gt`.

**Comment by pedrohcgs** [comment]
Hi, thanks for the detailed follow-up and apologies for the slow response!

**The short answer: yes, this is possible by working directly with the `att_gt()` output.**

The ATT(g,t) estimates, their (g,t) labels, and their influence functions are all accessible:

- `mp$att` — vector of ATT(g,t) estimates
- `mp$group`, `mp$t` — group and time labels for each entry
- `mp$inffunc` — N × K matrix of influence functions (one column per ATT(g,t))

**Equal-weight average over a custom subset:**

```r
# Select the (g,t) pairs you want
idx <- which((mp$group == g1 & mp$t == t1) |
             (mp$group == g1 & mp$t == t2) |
             (mp$group == g2 & mp$t == t2))

# Point estimate
att_custom <- mean(mp$att[idx])

# Influence function and standard error
inf_custom <- rowMeans(mp$inffunc[, idx])
se_custom  <- sqrt(mean(inf_custom^2) / length(inf_custom))
```

**A note on data-dependent weights (e.g., cohort-size weighted averages):**

If you use weights that are themselves functions of the data (e.g., proportional to cohort sizes), the influence function of the weighted average has two components:

```
IF_total = Σ w_g * IF_{ATT_g}  +  Σ ATT_g * IF_{w_g}
```

The first term is captured by the weighted combination of columns of `mp$inffunc`. The second term — the influence function of the weights — also needs to be included, otherwise standard errors will be understated. The internal function `get_agg_inf_func()` in our source handles this correctly and can serve as a reference for how to implement this properly.

Closing as the existing tools and the approach above cover this use case. Please reopen if you have further questions!
